### ISSUE TITLE
(fstat(STDOUT)) causes a parse error

### ISSUE THREAD

**Issue by thekid** [xp-framework/compiler#98]
Exception lang.ast.Errors (Errors { Expected ")", have ";" in braced at line 3, Missing semicolon after operator statement at line 3})

```diff
$ git diff
--- a/src/test/php/lang/ast/unittest/emit/BracesTest.class.php
+++ b/src/test/php/lang/ast/unittest/emit/BracesTest.class.php
@@ -93,6 +93,17 @@ class BracesTest extends EmittingTest {
     Assert::equals($expected, $r);
   }

+  #[Test]
+  public function function_call_in_braces() {
+    $r= $this->run('class <T> {
+      public function run() {
+        return false !== (fstat(STDOUT));
+      }
+    }');
+
+    Assert::true($r);
+  }
+
```

**Comment by thekid** [comment]
This is in the AST library, inside the following block:

```php
// This is ambiguous:
//
// - An expression surrounded by parentheses `($a ?? $b)->invoke()`;
// - A cast `(int)$a` or `(int)($a / 2)`.
//
// Resolve by looking ahead after the closing ")"
$this->prefix('(', 0, function($parse, $token) {
  ...
});
```

**Comment by thekid** [comment]
This fixes is but isn't satisfactory:

```diff
--- a/src/main/php/lang/ast/syntax/PHP.class.php
+++ b/src/main/php/lang/ast/syntax/PHP.class.php
@@ -222,7 +222,6 @@ class PHP extends Language {

     // This is ambiguous:
     //
-    // - An arrow function `($a) ==> $a + 1`
     // - An expression surrounded by parentheses `($a ?? $b)->invoke()`;
     // - A cast `(int)$a` or `(int)($a / 2)`.
     //
@@ -262,7 +261,14 @@ class PHP extends Language {
         $parse->forward();
         $parse->expecting('(', 'braced');
         $expr= $this->expression($parse, 0);
-        $parse->expecting(')', 'braced');
+
+        // See https://github.com/xp-framework/compiler/issues/98, unsure why we are
+        // short one ')' in this case though.
+        if (';' === $parse->token->value) {
+          $parse->queue= [$parse->token];
+        } else {
+          $parse->expecting(')', 'braced');
+        }
         return new Braced($expr, $token->line);
       }
     });
```

**Comment by thekid** [comment]
Fix released in https://github.com/xp-framework/ast/releases/tag/v7.0.1

**Comment by thekid** [comment]
Oops, this fix wasn't the right one after all!

```php
// This parses without problems
public function run() {
  $e= STDOUT;
  return false !== (fstat($e));
}

// This causes the problems
public function run() {
  return false !== (fstat(STDOUT));
}
```

**Comment by thekid** [comment]
**Really** fixed by https://github.com/xp-framework/ast/releases/tag/v7.0.1 - unittest included in XP Compiler, v6.1.0
